### PR TITLE
feat(query): auto extend datetime time part

### LINKED ISSUE #7528: Feature: timestamp auto extend minute second
**Summary**

support like this:

to_date('2022-02-02 ') -> 2022-02-02 00:00:00.000000

to_date('2022-02-02T') -> 2022-02-02 00:00:00.000000

to_date('2022-02-02 10') -> 2022-02-02 10:00:00.000000

to_date('2022-02-02 10:12') -> 2022-02-02 10:12:00.000000

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

support like this:

to_date('2022-02-02 ') -> 2022-02-02 00:00:00.000000

to_date('2022-02-02T') -> 2022-02-02 00:00:00.000000

to_date('2022-02-02 10') -> 2022-02-02 10:00:00.000000

to_date('2022-02-02 10:12') -> 2022-02-02 10:12:00.000000

Fixes #7528

### CODE REVIEW

**Comment by flaneur2020** [comment]
🐮 🐮 🐮

File: src/common/io/src/buffer/buffer_read_datetime_ext.rs [hunk 1]
```diff
@@ -97,15 +97,30 @@ where R: BufferRead
...
                 let size = self.keep_read(&mut buf, |f| (b'0'..=b'9').contains(&f))?;
-                match get_time(&mut buf, size) {
-                    Ok(time) => times.push(time),
-                    Err(e) => return Err(e),
-                }
-                if times.len() == 3 {
+                if size == 0 {
                     break;
+                } else {
+                    match get_time(&mut buf, size) {
```
Comment (BohuTANG) [inline src/common/io/src/buffer/buffer_read_datetime_ext.rs:103]: let time = get_time()?;
times.push(time);

Comment (TCeason) [reply src/common/io/src/buffer/buffer_read_datetime_ext.rs:103]: Got and there has a clippy err

```
error: it looks like the same item is being pushed into this Vec
   --> src/common/io/src/buffer/buffer_read_datetime_ext.rs:116:21
    |
116 |                     times.push(0);
    |                     ^^^^^
    |
    = note: `-D clippy::same-item-push` implied by `-D warnings`
    = help: try using vec![0;SIZE] or times.resize(NEW_SIZE, 0)
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#same_item_push

```

File: src/common/io/src/buffer/buffer_read_datetime_ext.rs [hunk 2]
```diff
@@ -97,15 +97,26 @@ where R: BufferRead
...
+                    let time = get_time(&mut buf, size)?;
+                    times.push(time);
+                    if times.len() == 3 {
+                        break;
+                    }
+                    self.ignore_byte(b':')?;
                 }
-                self.must_ignore_byte(b':')?;
             }
+            // Time part is HH:MM or HH
```
Comment (b41sh) [inline src/common/io/src/buffer/buffer_read_datetime_ext.rs:111]: better to add some comments and examples to explain this special case.

Comment (TCeason) [reply src/common/io/src/buffer/buffer_read_datetime_ext.rs]: Done

**Review by b41sh — APPROVED** [review]
LGTM

**Review by sundy-li — APPROVED** [review]

**Comment by TCeason** [comment]
Need modify doc.

**Pull request merged**